### DISCUSSION TITLE
Separating Values from Swim Meet Results PDF

### DISCUSSION THREAD

**Discussion by mkrieger95** [jsvine/pdfplumber#750] [Ask for help with specific PDFs]
Hi all,

This is probably very basic question, but having issues figuring out to break out the text in attached PDF results file for a swim meet.

Wanting to separate the values in similar order as they appear on the PDF.

Example data:

**Meet:** 2010 NC NMA Holiday Invit.
**Event:** Girls 8 & Under 25 Yard Freestyle
**Place:** 1
**First Name:** Katie
**Last Name:** Rauch
**Age:** 8
**Team:** ATOM-NC
**Result Time:** 15.95
**Points:** 20

This would ideally continue and would export to a CSV. Just can't figure out how to make code separate the values out.

Thank you in advance for your time and consideration to assist me.

[nc10134-results.pdf](https://github.com/jsvine/pdfplumber/files/9894021/nc10134-results.pdf)

**Comment by jsvine** [comment]
Hi @mkrieger95, and thanks for your interest in this library. Your PDF is a very interesting, and potentially tricky because of the various formatting styles it uses (e.g., for individual results vs. relays) and how the tables flow between columns. It should, however, be doable. The general steps I'd take:

- Use the `Page.lines` property to identify the lines on the page and their properties, then use that information with `Page.crop(...)` to isolate the two main halves (left and right) of the results.

- Use the remaining lines to detect the starting position of each sub-table, and `Page.crop(...)` again to focus on each, one at a time

- For the simple tables, you can probably use `Page.extract_table({ "horizontal_strategy": "text", "vertical_strategy": text" })` to get the table data itself ... but the more complex tables (such as those for the relays) may require you writing some custom logic, perhaps based on the output of `Page.extract_text(layout=True)` combined with regular expressions.
